File: capstone/src/main/java/impact/capstone/user/repository/UserRepository.java
package impact.capstone.user.repository;

import impact.capstone.model.entity.UserEntity;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.stereotype.Repository;

@Repository
public interface UserRepository extends JpaRepository<UserEntity, Long> {
    @Query("select (count(u) > 0) from UserEntity u where u.userId = ?1")
    boolean existsByUserId(String userId);
}

File: server/src/main/kotlin/deu/capstone/impact/server/repository/UserRepository.kt
package deu.capstone.impact.server.repository

import deu.capstone.impact.server.entity.user.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface UserRepository : JpaRepository<User, Long> {

    fun findByEmail(email: String): User?

}